### PR TITLE
[Fix][Convolution] Add dilation support

### LINKED ISSUE #1519: [FEAT][CONVOLUTION] support conv2d conv3d dilation
### Symptom / Motivation
Conv2dFwdOp, Conv2dBiasFwdOp, Conv3dFwdOp, and Conv3dBiasFwdOp expose dilation in the manifest-aligned constructor, but current implementation rejects dilation values other than 1.

### Root Cause Analysis
The op layer normalizes dilation and uses it for padding/output-size validation, then raises NotImplementedError when dilation is not all ones. The current Conv2d/Conv3d kernels do not receive dilation parameters and compute only the non-dilated convolution pattern.

### Related Files
- tileops/ops/convolution.py
- tileops/kernels/convolution.py
- tileops/manifest/convolution.yaml
- tests/ops/test_convolution.py
- benchmarks/ops/bench_convolution.py

## Goal
Implement dilated Conv2d/Conv3d semantics for bias and no-bias variants so the manifest dilation parameter is supported.

## Plan

1. Add dilation parameters to Conv2d/Conv3d kernel configuration and dispatch where needed.
2. Update kernel indexing to apply dilation along spatial dimensions.
3. Keep output shape validation aligned with PyTorch formulas.
4. Add correctness tests against torch.nn.functional.conv2d/conv3d for representative dilation values.
5. Re-run convolution tests and manifest validation.

## Constraints
- Keep dilation=1 behavior unchanged.
- Keep bias/no-bias operator variants separate.
- Do not promote manifest status to implemented until dilated cases pass correctness tests.

## Acceptance Criteria
- [ ] Modified files pass unit tests
- [ ] Conv2dFwdOp and Conv2dBiasFwdOp support dilation > 1
- [ ] Conv3dFwdOp and Conv3dBiasFwdOp support dilation > 1
- [ ] Tests compare dilated outputs with PyTorch references

### PR DESCRIPTION
**Pull request by RMLYC**
Closes #1519

## Summary

- Add dilation support to Conv2d/Conv2dBias and Conv3d/Conv3dBias kernel dispatch and indexing.
- Extend convolution tests with dilation cases in the existing Conv2dFixture and Conv3dFixture, without adding standalone dilation-only tests.
- Add model-derived dilation benchmark cases for DeepLabV3 ASPP Conv2d and 3D U-Net ASPP Conv3d.

## Test plan

- [x] `CUDA_VISIBLE_DEVICES=1 TMPDIR=/home/lyc/Project/TileOPs/.tmp/tvm python -m pytest tests/ops/test_convolution.py -q` (`45 passed in 8.75s`)
- [x] `CUDA_VISIBLE_DEVICES=1 TMPDIR=/home/lyc/Project/TileOPs/.tmp/tvm python -m pytest benchmarks/ops/bench_convolution.py -k "rate12 or rate6" -vvs` (`2 passed, 22 deselected in 8.05s`)
- [x] `python -m ruff check tests/ops/test_convolution.py benchmarks/ops/bench_convolution.py tileops/ops/convolution.py tileops/kernels/convolution.py`
- [x] `git diff --check -- tests/ops/test_convolution.py benchmarks/ops/bench_convolution.py tileops/ops/convolution.py tileops/kernels/convolution.py tileops/manifest/convolution.yaml`
- [x] `python scripts/validate_manifest.py --check-op Conv2dFwdOp` (passed with existing advisory warnings)
- [x] `python scripts/validate_manifest.py --check-op Conv3dFwdOp` (passed with existing advisory warnings)

## Benchmark

Run on GPU 1 with `CUDA_VISIBLE_DEVICES=1`; GPU reported by benchmark harness: NVIDIA H200. Ratio uses the repository benchmark convention: `torch_latency / tileops_latency`.

| Case | Source | TileOps latency | PyTorch baseline latency | Baseline / TileOps |
| --- | --- | ---: | ---: | ---: |
| `deeplabv3-aspp-3x3-rate12-fp16` | DeepLabV3/DeepLabV3+ ASPP 3x3 atrous conv on stride-16 encoder features | 0.2963 ms | 0.1273 ms | 0.43x |
| `3d-unet-aspp-3x3x3-rate6-fp16` | 3D U-Net + 3D ASPP medical segmentation branch | 0.0970 ms | 0.5258 ms | 5.42x |

## Regression

- Conv2d no-bias regression now covers `dilation=2` against `torch.nn.functional.conv2d`.
- Conv3d no-bias regression now covers `dilation=2` against `torch.nn.functional.conv3d`.

### CODE REVIEW

**Review by superAngGao — APPROVED** [review]
Looks good to me. The dilation support is threaded consistently through the op constructors, kernel indexing/output-shape math, fake wrappers, manifest rules, tests, and benchmarks. I did not find any blocking issues.

**Review by lcy-seso — APPROVED** [review]
